Signal/src/call/CallService.swift

```diff
@@ -300,32 +305,42 @@ protocol CallServiceObserver: class {
 
             return self.peerConnectionClient!.createOffer()
         }.then { (sessionDescription: HardenedRTCSessionDescription) -> Promise<Void> in
+            guard self.call == call else {
+                return Promise(error: CallError.obsoleteCall(description:"obsolete call in \(#function)"))
+            }
+
             return self.peerConnectionClient!.setLocalSessionDescription(sessionDescription).then {
                 let offerMessage = OWSCallOfferMessage(callId: call.signalingId, sessionDescription: sessionDescription.sdp)
                 let callMessage = OWSOutgoingCallMessage(thread: thread, offerMessage: offerMessage)
                 return self.messageSender.sendCallMessage(callMessage)
             }
         }.then {
+            guard self.call == call else {
+                return Promise(error: CallError.obsoleteCall(description:"obsolete call in \(#function)"))
```

Comment:
nit: idiomatic to throw the error rather than to return a failed promise (here and above).